server/base/src/main/java/org/apache/accumulo/server/mem/LowMemoryDetector.java

```diff
@@ -135,48 +134,55 @@ public void logGCInfo(AccumuloConfiguration conf) {
         prevGcTime.put(gcBean.getName(), time);
       }
 
-      long mem = rt.freeMemory();
-      if (maxIncreaseInCollectionTime == 0) {
-        gcTimeIncreasedCount = 0;
-      } else {
-        gcTimeIncreasedCount++;
-        if (gcTimeIncreasedCount > 3 && mem < rt.maxMemory() * freeMemoryPercentage) {
+      Runtime rt = Runtime.getRuntime();
+      final long maxConfiguredMemory = rt.maxMemory();
+      final long allocatedMemory = rt.totalMemory();
+      final long allocatedFreeMemory = rt.freeMemory();
+      final long freeMemory = maxConfiguredMemory - (allocatedMemory - allocatedFreeMemory);
+      final double lowMemoryThreshold = maxConfiguredMemory * freeMemoryPercentage;
```

Comment:
Should cast this to long here, so it doesn't need to be done three times later.

Reply:
Addressed in 3ed08fc601